raiden/raiden_service.py

```diff
@@ -134,6 +136,20 @@ def save_snapshot(serialization_file, raiden):
         )
 
 
+def endpoint_registry_exception_handler(greenlet):
+    try:
+        greenlet.get()
+    except (JSONRPCPollTimeoutException, Exception) as e:
+        if (e.message == "timeout when polling for transaction" or
+           isinstance(e, JSONRPCPollTimeoutException)):
+                log.fatal("Endpoint registry failed: %s. "
+                          "Ethereum RPC API might be unreachable.",
+                          repr(e))
+        else:
+            log.fatal("Endpoint registry failed: %s. ", repr(e))
```

Comment:
You can use `log.fatal("bla", exc_info=True)` to have the logging entry automatically include the exception and traceback (which would be useful here since this is the branch for unexpected exceptions).